packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -533,6 +509,29 @@ export interface ProjectProps extends CommonProjectProps {
  * A representation of a CodeBuild Project.
  */
 export class Project extends ProjectBase {
+
+  public static fromProjectArn(scope: Construct, id: string, projectArn: string): IProject {
+    class Import extends ProjectBase {
+      public readonly grantPrincipal: iam.IPrincipal;
+      public readonly projectArn = projectArn;
+      public readonly projectName = scope.node.stack.parseArn(projectArn).resourceName!;
+      public readonly role?: iam.Role = undefined;
+
+      constructor(s: Construct, i: string) {
+        super(s, i);
+        this.grantPrincipal = new iam.ImportedResourcePrincipal({ resource: this });
+      }
+
+      public export(): ProjectAttributes {
```

Comment:
We don't need `export()` any more, right? Might as well get rid of it immediately.

Reply:
I am going to punt that to a subsequent PR (just reduce the amount of changes). It's overwhelming.